<?xml version="1.0"?>
<launch>
  <arg name="subscript" default="" />
  <arg name="enable_logging" default="false" />
  <arg name="enable_ground_truth" default="true" />
  <arg name="param_file" default="$(find rrt_explorer)/resource/exploration.yaml" />
  
   <arg name="tf_frame" default="world" />
   <arg name="resolution" default="0.15" /> 
   <arg name="visualize_max_z" default="3" />
   <arg name="sensor_max_range" default="3.0" />
   <arg name="map_publish_frequency" default="1.0"/>
   <arg name="probability_hit" default="0.7" />
   <arg name="probability_miss" default="0.4" />
   <arg name="threshold_min" default="0.12" />
   <arg name="threshold_max" default="0.97" />
   <arg name="threshold_occupancy" default="0.7" />
   
  <arg name="pcl_topic" default="/current_viewpoint_pointcloud_transfered" />
  <arg name="stl_file_path" default="" />
  <arg name="mesh_resolution" default="1.0" />
 <arg name="treat_unknown_as_occupied" default="false" />

    <node name="rrt_planner" pkg="rrt_explorer" type="rrt_planner" output="screen">
      <param name="tf_frame" type="string" value="$(arg tf_frame)" />
      <param name="resolution" type="double" value="$(arg resolution)" />
      <param name="visualize_max_z" type="double" value="$(arg visualize_max_z)" />
      <param name="sensor_max_range" type="double" value="$(arg sensor_max_range)" />
      <param name="stl_file_path" type="string" value="$(arg stl_file_path)" />
      <param name="mesh_resolution" type="double" value="$(arg mesh_resolution)" />
      <param name="map_publish_frequency" value="$(arg map_publish_frequency)" />
      <remap from="pointcloud_throttled" to="$(arg pcl_topic)"/> 
      <remap from="poseStamped" to="/current_pose"/>
      <!--remap from="pose" to="/$(arg mav_name)$(arg subscript)/ground_truth/pose_with_covariance"/-->
      <!--remap from="pose" to="/$(arg mav_name)$(arg subscript)/mavros/local_position/pose"/-->
      <remap from="rrt_explorer" to="/rrt_explorer"/>
      <rosparam command="load" file="$(arg param_file)" />

    </node>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find rrt_explorer)/rviz/rrt_planner.rviz"/>
</launch>
